core/infra/connection/connection_with_grpc.cpp

```diff
@@ -0,0 +1,267 @@
+        throw "No implements";
+    }
+
+    template<>
+    connection_object::ConsensusEvent encodeConsensusEvent(
+        std::unique_ptr<
+            ConsensusEvent<
+                Transaction<
+                    Add<Asset>
+                >
+            >
+        >&& event
+    ) {
+        connection_object::Asset asset;
+        auto txObj = event->dump().dictSub["transaction"];
+        auto assetObj = txObj.dictSub["command"].dictSub["object"];
+
+        asset.set_domain(assetObj.dictSub["domain"].str);
+        asset.set_name(assetObj.dictSub["name"].str);
+        asset.set_value(static_cast<google::protobuf::uint64>(assetObj.dictSub["value"].integer));
+        asset.set_precision(static_cast<google::protobuf::uint64>(assetObj.dictSub["precision"].integer));
+
+        connection_object::Transaction tx;
+        tx.set_type(event->getCommandName());
+        tx.set_senderpubkey(event->dump().dictSub["transaction"].dictSub["senderPublicKey"].str);
+        tx.mutable_asset()->CopyFrom(asset);
+
+        connection_object::ConsensusEvent consensusEvent;
+        consensusEvent.mutable_transaction()->CopyFrom(tx);
+
+        if(!event->eventSignatures().empty()) {
+            for (auto &esig: event->eventSignatures()) {
+                connection_object::EventSignature eventSig;
+                eventSig.set_publickey(std::get<0>(esig));
+                eventSig.set_signature(std::get<1>(esig));
+                consensusEvent.add_eventsignatures()->CopyFrom(eventSig);
+            }
+        }
+
+        return consensusEvent;
+    }
+
+    template<typename T>
+    T decodeConsensusEvent(const connection_object::ConsensusEvent& event){
+        logger::error("connection","No implements error :"+ std::string(typeid(T).name()));
+        throw "No implements";
+    }
+
+    template<>
+    std::unique_ptr<
+        ConsensusEvent<
+            Transaction<
+                Add<Asset>
+            >
+        >
+    > decodeConsensusEvent(
+        const connection_object::ConsensusEvent& event
+    ) {
+        auto tx = event.transaction();
+        auto asset = tx.asset();
+
+        auto consensusEvent =  std::make_unique<ConsensusEvent<
+            Transaction<
+                    Add<Asset>
+            >
+        >>(
+            tx.senderpubkey(),
+            asset.domain(),
+            asset.name(),
+            asset.value(),
+            asset.precision()
+        );
+        for(const auto& esig: event.eventsignatures()){
+            consensusEvent->addSignature(esig.publickey(), esig.signature());
+        }
+        for(const auto& txsig: event.transaction().txsignatures()){
+            consensusEvent->addTxSignature(txsig.publickey(), txsig.signature());
+        }
+        return consensusEvent;
+    }
+
+    class IrohaConnectionClient {
+        public:
+        IrohaConnectionClient(std::shared_ptr<Channel> channel)
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class 'IrohaConnectionClient' has a constructor with 1 argument that is not explicit.](https://www.codacy.com/app/soramitsu/iroha/file/4621990420/issues/source?bid=3712501&fileBranchId=3957673#l152)